src/components/pages/comparison.js

```diff
@@ -0,0 +1,315 @@
+import by from 'thenby';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import DiffView from '../diff-view.js';
+import routes from '../../utils/routes.js';
+import { Button, Select, Tooltip } from 'antd';
+import SectionDisplay from '../section-display.js';
+import { useRequest } from '../request-context.js';
+import { useService } from '../container-context.js';
+import { useDateFormat } from '../locale-context.js';
+import { AlignLeftOutlined } from '@ant-design/icons';
+import { Trans, useTranslation } from 'react-i18next';
+import PreviewIcon from '../icons/general/preview-icon.js';
+import PluginRegistry from '../../plugins/plugin-registry.js';
+import React, { Fragment, useEffect, useMemo, useState } from 'react';
+import { documentRevisionShape } from '../../ui/default-prop-types.js';
+import { ensurePluginComponentAreLoadedForSections } from '../../utils/plugin-utils.js';
+import { createDocumentRevisionComparison, SECTION_CHANGE_TYPE } from '../../utils/document-revision-comparison-utils.js';
+
+const { Option } = Select;
+
+const NOOP = () => {};
+
+const SECTION_CONTEXT = { isPreview: true };
+
+const DISPLAY_MODE = {
+  text: 'text',
+  preview: 'preview'
+};
+
+class ComparisonPreloader {
+  static dependencies = [PluginRegistry];
+
+  constructor(pluginRegistry) {
+    this.pluginRegistry = pluginRegistry;
+  }
+
+  preload({ initialState }) {
+    return ensurePluginComponentAreLoadedForSections({
+      sections: initialState.revisions.flatMap(revision => revision.sections),
+      pluginRegistry: this.pluginRegistry,
+      displayOnly: true
+    });
+  }
+}
+
+function Comparison({ initialState, PageTemplate }) {
+  const request = useRequest();
+  const { revisions } = initialState;
+  const { formatDate } = useDateFormat();
+  const { t } = useTranslation('comparison');
+  const pluginRegistry = useService(PluginRegistry);
+  const [comparison, setComparison] = useState(null);
+  const [oldRevisionId, setOldRevisionId] = useState(request.query.oldId);
+  const [newRevisionId, setNewRevisionId] = useState(request.query.newId);
+  const [displayModesByMappedRevisionKey, setDisplayModesByMappedRevisionKey] = useState({});
+
+  const mappedRevisions = useMemo(() => [...revisions.sort(by(x => x.order, 'desc'))].map((revision, index) => {
+    const isLatestVersion = index === 0;
+    const version = revisions.length - index;
+    const latestVersionText = isLatestVersion ? ` (${t('common:latest')})` : '';
+    const versionText = `${t('common:version')} ${version}${latestVersionText}`;
```

Comment:
Since we now have basically the exact same logic here and on `document-version-history`, but also a variation in the `getDocumentRevisionHistoryVersionInfo` helper used on the `document` page (which is still sorting incorrectly by `createdOn` and not by `order` 😬 ), I'm thinking we could transform the helper into a `getDocumentRevisionsWithVersionInfo` (or something of sorts) that maps `isLatestVersion` and `versionText` to the `documentRevision`. And use that in all 3 places. 
What do you think?

Reply:
Sounds like a good idea! I'll extract a utility function.

Reply:
Done!

Reply:
Perfect! 👏